diff --git a/src/libplctag/protocols/ab/eip_cip.c b/src/libplctag/protocols/ab/eip_cip.c
index eab8ade..c57e570 100644
--- a/src/libplctag/protocols/ab/eip_cip.c
+++ b/src/libplctag/protocols/ab/eip_cip.c
@@ -212,6 +212,8 @@ int tag_tickler(plc_tag_p tag_arg) {
     rc = check_request_status(tag);
     if(rc != PLCTAG_STATUS_OK) { return rc; }
 
+    pdebug(DEBUG_DETAIL, "Tag read in progress: %d", tag->read_in_progress);
+    pdebug(DEBUG_DETAIL, "Tag write in progress: %d", tag->write_in_progress);
     if(tag->read_in_progress) {
         if(tag->use_connected_msg) {
             rc = check_read_status_connected(tag);
diff --git a/src/libplctag/protocols/ab/session.c b/src/libplctag/protocols/ab/session.c
index 0a33cec..839d25e 100644
--- a/src/libplctag/protocols/ab/session.c
+++ b/src/libplctag/protocols/ab/session.c
@@ -1625,8 +1625,6 @@ int process_requests(ab_session_p session) {
     critical_block(session->session_mutex) {
         int available_payload = session_get_available_cip_payload_space(session);
 
-        pdebug(DEBUG_DETAIL, "Available payload space is %d bytes.", available_payload);
-
         // FIXME - no logging in a mutex!
         // pdebug(DEBUG_DETAIL, "FIXME: available payload space %d", available_payload);
 
